temporalio/client.py

```diff
@@ -188,10 +198,80 @@ def data_converter(self) -> temporalio.converter.DataConverter:
         """Data converter used by this client."""
         return self._config["data_converter"]
 
+    @overload
+    async def start_workflow(
+        self,
+        workflow: Callable[[WorkflowClass], Awaitable[WorkflowReturnType]],
+        *,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: temporalio.common.WorkflowIDReusePolicy = temporalio.common.WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
+        header: Optional[Mapping[str, Any]] = None,
+        start_signal: Optional[str] = None,
+        start_signal_args: Iterable[Any] = [],
+    ) -> WorkflowHandle[WorkflowClass, WorkflowReturnType]:
+        ...
+
+    @overload
+    async def start_workflow(
+        self,
+        workflow: Callable[
+            [WorkflowClass, LocalParamType], Awaitable[WorkflowReturnType]
+        ],
+        arg: LocalParamType,
+        *,
+        id: str,
+        task_queue: str,
+        execution_timeout: Optional[timedelta] = None,
+        run_timeout: Optional[timedelta] = None,
+        task_timeout: Optional[timedelta] = None,
+        id_reuse_policy: temporalio.common.WorkflowIDReusePolicy = temporalio.common.WorkflowIDReusePolicy.ALLOW_DUPLICATE,
+        retry_policy: Optional[temporalio.common.RetryPolicy] = None,
+        cron_schedule: str = "",
+        memo: Optional[Mapping[str, Any]] = None,
+        search_attributes: Optional[Mapping[str, Any]] = None,
+        header: Optional[Mapping[str, Any]] = None,
+        start_signal: Optional[str] = None,
+        start_signal_args: Iterable[Any] = [],
+    ) -> WorkflowHandle[WorkflowClass, WorkflowReturnType]:
+        ...
+
+    @overload
     async def start_workflow(
         self,
         workflow: str,
-        *args: Any,
+        arg: Any = temporalio.common._arg_unset,
+        *,
+        args: Iterable[Any] = [],
```

Comment:
Isn't there a way to get typed `args` or at least support `args` with a `Callable` instead of a string?

Reply:
What about `Callable` with `args: Iterable[Any]`?

Reply:
Oh, you're saying have an overload for untyped arg form of function reference? I can do that. I was basically forcing those users to fall back to string usage as a way of discouraging multi-argument workflows, but I have no problem adding that overload if we want it.

Reply:
Yeah that's what I meant, I agree we want to encourage single param but I wouldn't take it that far

Reply:
:+1: Will add overload in all places (there's about 12 of them) for untyped multi-args with function references.

Reply:
Overloads added